File: app/src/main/java/tech/salroid/filmy/parser/SearchResultParseWork.java
package tech.salroid.filmy.parser;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import java.util.ArrayList;
import java.util.List;
import tech.salroid.filmy.data.SearchData;

public class SearchResultParseWork {

    private String result;

    public SearchResultParseWork(String result) {
        this.result = result;
    }

    public List<SearchData> parseSearchData() {
        final List<SearchData> searchArray = new ArrayList();
        SearchData searchData;

        try {
            JSONObject jsonobject = new JSONObject(result);
            JSONArray jsonArray = jsonobject.getJSONArray("results");

            for (int i = 0; i < jsonArray.length(); i++) {

                String id = jsonArray.getJSONObject(i).getString("id");
                String date = jsonArray.getJSONObject(i).getString("release_date");
                String moviePoster = "http://image.tmdb.org/t/p/w185" + jsonArray.getJSONObject(i).getString("poster_path");
                String title = jsonArray.getJSONObject(i).getString("original_title");
                String type = "movie";

                searchData = new SearchData(id);

                if (!(moviePoster.equals("null") && date.equals("null"))) {
                    searchData.setDate(date);
                    searchData.setType(type);
                    searchData.setMovie(title);
                    searchData.setPoster(moviePoster);
                    searchArray.add(searchData);
                }
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return searchArray;
    }
}

File: app/src/main/java/tech/salroid/filmy/parser/SearchResultParseWork.kt
package tech.salroid.filmy.parser

import org.json.JSONException
import org.json.JSONObject
import tech.salroid.filmy.data.SearchData
import java.util.*

class SearchResultParseWork(private val result: String) {

    fun parseSearchData(): List<SearchData> {
        val searchArray = ArrayList<SearchData>()

        var searchData: SearchData

        try {
            val jsonObject = JSONObject(result)
            val jsonArray = jsonObject.getJSONArray("results")

            for (i in 0 until jsonArray.length()) {
                val id = jsonArray.getJSONObject(i).getString("id")
                val date = jsonArray.getJSONObject(i).getString("release_date")
                val moviePoster = "http://image.tmdb.org/t/p/w185" + jsonArray.getJSONObject(i)
                    .getString("poster_path")
                val title = jsonArray.getJSONObject(i).getString("original_title")
                val type = "movie"

                searchData = SearchData(id)
                if (!(moviePoster == "null" && date == "null")) {
                    searchData.date = date
                    searchData.type = type
                    searchData.movie = title
                    searchData.poster = moviePoster
                    searchArray.add(searchData)
                }
            }

        } catch (e: JSONException) {
            e.printStackTrace()
        }
        return searchArray
    }
}